### DISCUSSION TITLE
add is `Prop:isphygunned`, `Prop:isfrozen`, `Prop:ispropgravgunned` etc

### DISCUSSION THREAD

**Discussion by DecVes** [wiremod/wire#3053] [Suggestions]
This should be easy to implement and its extremely useful for debugging and some other things. the is `Prop:ispropphysgunned` only occurs when a prop is actively being held by a physgun, you can also add this for grav gun `Prop:ispropgravgunned`. and `Prop:isfrozen` is quite self explanitory.

I would hope that these functions can be used in `if` statements

**Comment by DecVes** [comment]
i think this is a very good suggestion

**Comment by Denneisk** [comment]
Sounds doable. Exposing `GM:OnPhysgunPickup` and `GM:GravGunOnPickedUp` needs to be done eventually, too, I guess. Having specific functions for it sounds a little tacky but maybe it could be added.

There already is an `entity:isFrozen()` function for what you described, however.

**Reply by DecVes** [reply]
i havent seen `Prop:isfrozen()` unless i just didnt see it but nonetheless this addition for `Prop:OnphysgunPickup` and `Prop:GravGunOnPickedUp` seems very good to have for debugging and other things you might do with them
